### PR TITLE
Refactor [Requires]IO

### PR DESCRIPTION
**Pull request by TagnumElite**

### CODE REVIEW

**Comment by chris48s** [comment]
btw, you might have already noticed but all the work you did recently on adding new badges for steam and wordpress is now deployed on https://shields.io :rocket:

File: lib/all-badge-examples.js [hunk 1]
```diff
@@ -1494,10 +1490,6 @@ const allBadgeExamples = [
         title: 'Puppet Forge',
         previewUrl: '/puppetforge/mc/camptocamp.svg',
       },
-      {
-        title: 'Maintenance',
-        previewUrl: '/maintenance/yes/2017.svg',
-      },
```
[the remark this answers is not in the document]
Comment (TagnumElite) [reply lib/all-badge-examples.js]: Oh, yeah. I was planning to do maintenance and RequiresIO in one pull request. Left this in by mistake, or I could just refactor maintenance as well. I am gonna refactor maintenance as well.

File: services/requires/requires.service.js [hunk 2]
```diff
@@ -1,53 +1,70 @@
...
+const statusSchema = Joi.object({
+  status: Joi.string().required(),
+}).required()
+
+module.exports = class RequiresIo extends BaseJsonService {
+  static get url() {
+    return {
+      base: 'requires',
+      format: '([^/]+/[^/]+/[^/]+)(?:/(.+))?',
+      capture: ['repo', 'branch'],
```
[the remark this answers is not in the document]
Comment (TagnumElite) [reply services/requires/requires.service.js]: That's true, but what if requires.io support gitlab or something else at another point. This way it won't need to rewritten.

File: lib/all-badge-examples.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (chris48s) [reply lib/all-badge-examples.js]: Oh great, thanks for volunteering to take that one on.

Lets make the maintenance changes in a different PR to the requires.io stuff. It is much easier to review smaller chunks of work :)

File: services/requires/requires.service.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (chris48s) [reply services/requires/requires.service.js]: If you wanted to make it more flexible, you could capture `([^/]+)/([^/]+/[^/]+)` instead.. the main thing is to make it clearer what information we're capturing from the regex. Will leave it up to you how you want to do it.

**Pull request merged**